pkg/utils/configMapUtils.go

```diff
@@ -30,14 +31,40 @@ func (clientSets ClientSets) ValidateConfigMap(configMapName string, experiment
 }
 
 // SetConfigMaps sets the value of configMaps in Experiment Structure
-func (expDetails *ExperimentDetails) SetConfigMaps(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetConfigMaps(clients ClientSets, engineName string) {
 
 	chaosExperimentObj, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
-		log.Infof("Unable to get ChaosEXperiment Resource, wouldn't not be able to patch ConfigMaps")
+		log.Infof("Unable to get ChaosExperiment Resource, wouldn't be able to patch ConfigMaps")
+	}
+	experimentConfigMaps := chaosExperimentObj.Spec.Definition.ConfigMaps
+
+	chaosEngineObj, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosEngines(expDetails.Namespace).Get(engineName, metav1.GetOptions{})
+	if err != nil {
+		log.Infof("Unable to patch ConfigMaps, error: %v", err)
 	}
-	configMaps := chaosExperimentObj.Spec.Definition.ConfigMaps
-	expDetails.ConfigMaps = configMaps
+	experimentsList := chaosEngineObj.Spec.Experiments
+	for i := range experimentsList {
```

Comment:
Can we a write a separate function for getting the index of the current Experiment from the list of experiment, something like `checkExperimentListForExp([]ExperimentList, ExperimentName`

Reply:
Please take a look again.